nupic/research/frameworks/pytorch/datasets/common.py

```diff
@@ -0,0 +1,80 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2020, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import os
+
+from torchvision import datasets, transforms
+
+DATASETS_STATS = {
+    "ImageNet": ((0.485, 0.456, 0.406), (0.229, 0.224, 0.225)),
+    "TinyImageNet": ((0.485, 0.456, 0.406), (0.229, 0.224, 0.225)),
+    "CIFAR10": ((0.4914, 0.4822, 0.4465), (0.2023, 0.1994, 0.2010)),
+    "CIFAR100": (
+        (0.50707516, 0.48654887, 0.44091784),
+        (0.26733429, 0.25643846, 0.27615047),
+    ),
+    "MNIST": ((0.13062755,), (0.30810780,)),
+}
+
+
+class TorchvisionDatasetManager(object):
+
+    def __init__(self, data_dir, dataset_name="MNIST"):
+
+        # TODO: calculate statistics for any torchvision dataset, if not available
+        if dataset_name not in DATASETS_STATS.keys():
+            raise ValueError(f"{dataset_name} not available.")
+        transform = self.base_transform(*DATASETS_STATS[dataset_name])
+
+        # TODO: rename data to dataset_dir
+        dataset_class = getattr(datasets, dataset_name)
+        self.train_dataset = dataset_class(
+            root=os.path.expanduser(data_dir),
+            train=True,
+            transform=transform,
+            download=False,
+        )
+
+        self.val_dataset = dataset_class(
+            root=os.path.expanduser(data_dir),
+            train=False,
+            transform=transform,
+            download=False,
+        )
+
+    def get_train_dataset(self):
+        return self.train_dataset
+
+    def get_val_dataset(self):
+        return self.val_dataset
```

Comment:
Could `SupervisedExperiment.create_validation_dataloader` call `DatasetManager.create_validation_dataset` directly? (To solve the "test script" problem I mention in a different comment)

In fact, is there a reason to ever need to instantiate a DatasetManager? It's a bit of a red flag that we're instantiating managers then never storing them. Maybe these classmethods should just be functions that return a dataset -- is there a reason we need a dedicated "Manager" class? This seems like a job for factory-style functions, similar to how the `common_layers.py` contains factory functions for layers. (Conveniently, each of those hypothetical functions could simply be called a "dataset" since they are dataset constructors.)

Reply:
This is good feedback. I guess there isn't a need for a whole datamanager class. I've refactored things to call a `create_datasets_func`. I returns the train and val datasets (and a test set if any).